### PR TITLE
Pul request for code review

### PR DESCRIPTION
**Pull request by dili021**
-Made two versions of the bubble_sort method, one for numbers and one for strings, bubble_sort and bubble_sort_by, respectively.
-Fixed all rubocop offenses.
-Added some \n (newlines) to make the output cleaner and a counter to display the number of sorting rounds.
*post code review:
-Removed print commands and counter from method as requested by the TSE.

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+def bubble_sort_by(arr)
+  print 'Your array before sorting: ', arr, "\n\n"
```
Comment (shshamim63) [inline bubble_sort.rb]: Delete the print. sort does not print anything.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  until sorted
+    sorted = true
+    (arr.length - 1).times do |i|
+      if yield(arr[i], arr[i + 1]).positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        sorted = false
+      end
+    end
+    counter += 1
+    print "round #{counter} of sorting: ", arr, "\n\n"
```
Comment (shshamim63) [inline bubble_sort.rb]: Delete the print also.

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  end
+  print 'Your array is now sorted!'
+  puts
```
Comment (shshamim63) [inline bubble_sort.rb]: instead of those two lines place `arr`

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+end
+bubble_sort_by(%w[hello dfhadfhad asg asdgafhafdhadheh dasfhadfhdafhafghafh as]) do |left, right|
+  left.length - right.length
+end
+
+def bubble_sort(arr)
+  print 'Your array before sorting: ', arr, "\n\n"
```
Comment (shshamim63) [inline bubble_sort.rb]: No need to print a message.

File: bubble_sort.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  until sorted
+    sorted = true
+    (arr.length - 1).times do |i|
+      if arr[i] > arr[i + 1]
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        sorted = false
+      end
+    end
+    counter += 1
+    print "round #{counter} of sorting: ", arr, "\n\n"
```
Comment (shshamim63) [inline bubble_sort.rb]: Delete the print please.

File: bubble_sort.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  end
+  print 'Your array is now sorted!'
+  puts
```
Comment (shshamim63) [inline bubble_sort.rb]: Place `arr` instead of the marked two lines.

**Review by shshamim63 — COMMENTED** [review]
## Status :space_invader: [Changes Required] :space_invader: 
___
Awesome job so far :clap: . Just figured out some issues that are holding me to approve this project now. Those issues are simple and I am pretty much confident that you will cover them easily. I will try my best to comment those issues based on logic. Let me know if you have different angel of view.
___
### :spades: Mandatory Changes :spades: 
___
1. The purpose of a sorting is to accept an array, on which it will perform action based on the provided condition. At the end it will return the modified array. But on your one I can see some outputs. That is actually against the basic functionality of a sort. Try the builtin sort of ruby `[3, 2, 1].sort` which only returns a sorted array. . So lets modify the code that you have written. Follow the comments to fix the issues.
___
#### Please fix the mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @dili021 and @bouaik  :balloon: :lollipop: 

### Your project works well,  and you've implemented all the changes requested by the previous reviewer, I won't keep you here any longer, you are good to go this project is approved!! :tada: :heavy_check_mark:  

Happy coding! :woman_technologist: 

PROJECT STATUS: APPROVED :green_apple:

**Pull request merged**